summarize/src/analysis.rs

```diff
@@ -159,33 +159,47 @@ pub fn perform_analysis(data: ProfilingData) -> Results {
                     thread.stack.pop();
                 }
 
+                let current_event_duration = current_event.duration().unwrap();
+
                 // If there is something on the stack, subtract the current
                 // interval from it.
                 if let Some(current_top) = thread.stack.last() {
                     record_event_data(&current_top.label, &|data| {
-                        data.self_time -= current_event.duration().unwrap();
+                        match &current_top.event_kind[..] {
+                            QUERY_EVENT_KIND | GENERIC_ACTIVITY_EVENT_KIND => {
+                                data.self_time -= current_event_duration;
+                            }
+                            INCREMENTAL_LOAD_RESULT_EVENT_KIND => {
+                                data.incremental_load_time  -= current_event_duration;
+                            }
+                            _ => {
```

Comment:
We probably need to handle `QUERY_BLOCKED_EVENT_KIND` here.

Reply:
Hm, that definitely shouldn't have anything on top of it, right?

Reply:
I'm not sure I know enough about how the query system works when parallelism is turned on to say. Does the thread stay blocked when a query is blocked or does it try to work on another query?

Reply:
It stays blocked.

Reply:
Ah ok. You're right then.